assets/js/googlesitekit/data/create-existing-tag-store.js

```diff
@@ -0,0 +1,189 @@
+
+const { createRegistryControl, createRegistrySelector } = Data;
+
+// Actions
+const FETCH_GET_EXISTING_TAG = 'FETCH_GET_EXISTING_TAG';
+const RECEIVE_GET_EXISTING_TAG = 'RECEIVE_GET_EXISTING_TAG';
+const WAIT_FOR_EXISTING_TAG = 'WAIT_FOR_EXISTING_TAG';
+
+/**
+ * Creates a store object that includes actions and selectors for getting existing tags.
+ *
+ * @since n.e.x.t
+ * @private
+ *
+ * @param {string}   type                 The data to access. One of 'core' or 'modules'.
+ * @param {string}   identifier           The data identifier, eg. a module slug like 'analytics'.
+ * @param {Object}   options              Options to consider for the store.
+ * @param {Array}    options.tagMatchers  The tag matchers used to extract tags from HTML.
+ * @param {Function} [options.isValidTag] Optional. Function to detect whether a tag is valid or not.
+ * @param {string}   [options.storeName]  Optional. Store name to use. Default is '{type}/{identifier}'.
+ * @return {Object} The existing tag store object, with additional `STORE_NAME` and
+ * INITIAL_STATE` properties.
+ */
+export const createExistingTagStore = ( type, identifier, {
+	tagMatchers,
+	isValidTag = ( tag ) => typeof tag === 'string' && tag.length > 0,
+	storeName = undefined,
+} = {} ) => {
+	invariant( type, 'type is required.' );
+	invariant( identifier, 'identifier is required.' );
+	invariant( 'function' === typeof isValidTag, 'isValidTag must be a function.' );
+	invariant( Array.isArray( tagMatchers ), 'tagMatchers must be an Array.' );
+
+	const STORE_NAME = storeName || `${ type }/${ identifier }`;
+
+	const INITIAL_STATE = {
+		existingTag: undefined,
+	};
+
+	const actions = {
+		fetchGetExistingTag() {
+			return {
+				payload: {},
+				type: FETCH_GET_EXISTING_TAG,
+			};
+		},
+		receiveGetExistingTag( existingTag ) {
+			invariant(
+				existingTag !== undefined && ( existingTag === null || isValidTag( existingTag ) ),
+				'existingTag must be a valid tag or null.'
+			);
+
+			return {
+				payload: { existingTag },
+				type: RECEIVE_GET_EXISTING_TAG,
+			};
+		},
+		*waitForExistingTag() {
+			yield {
+				payload: {},
+				type: WAIT_FOR_EXISTING_TAG,
+			};
+		},
+	};
+
+	const controls = {
+		[ FETCH_GET_EXISTING_TAG ]: createRegistryControl( ( registry ) => async () => {
+			const homeURL = registry.select( CORE_SITE ).getHomeURL();
+			const ampMode = registry.select( CORE_SITE ).getAMPMode();
+			const existingTagURLs = await getExistingTagURLs( { homeURL, ampMode } );
+
+			for ( const url of existingTagURLs ) {
+				await registry.dispatch( CORE_SITE ).waitForHTMLForURL( url );
+				const html = registry.select( CORE_SITE ).getHTMLForURL( url );
+				const tagFound = extractExistingTag( html, tagMatchers );
+				if ( tagFound ) {
+					return tagFound;
+				}
+			}
+
+			return null;
+		} ),
+		[ WAIT_FOR_EXISTING_TAG ]: createRegistryControl( ( registry ) => () => (
+			registry.__experimentalResolveSelect( STORE_NAME ).getExistingTag()
```

Comment:
I'm not sure where it was decided to use this function. It's currently causing one of the existing Jest tests to fail: https://travis-ci.com/github/google/site-kit-wp/jobs/363549318
Maybe it's because we don't resolve selectors when passing data to them (related: #1655)? cc @aaemnnosttv @tofumatt 

Either way, let's revert back to how this originally worked (in the Analytics store implementation from before) - with that tests all pass, and also maybe we shouldn't be using functions marked experimental in Gutenberg.